### ISSUE TITLE
Transfering community ownership

### ISSUE THREAD

**Issue by momothereal** [withspectrum/spectrum#3859]
**Issue Type**

- [ ] Bug Report
- [x] Feature Idea
- [ ] Technical Discussion
- [ ] Question (these will be auto-closed, please ask them on Spectrum instead https://spectrum.chat/spectrum/open)

There should be a way to transfer the ownership of a community to another server member. If #3749 is completed, an owner who wishes to delete their account should have a way to transfer the community to someone else.

**Comment by brianlovin** [comment]
Agreed - we've been doing this manually for a bunch of folks and it's starting to happen enough that we should just build the UI controls for it :P

**Comment by momothereal** [comment]
I'd like to take this as a challenge :)

From a quick glance, it looks like a simple permission transfer (`isOwner`), as well as updating the administrator email and Stripe ID. Am I on the right path?

**Comment by brianlovin** [comment]
@momothereal loosely, yep
- change `isOwner` field on the correct `usersCommunities` records
- change `isOwner` field on the correct `usersChannels` records
- change `administratorEmail` and `creatorId`

One thing I haven't thought through completely right now is how to properly handle the transfer of a community that is paying for features, and how to ensure that the new owner has their own Stripe billing set up and the old owner's card doesn't get stuck as the payment method.

**Comment by momothereal** [comment]
It could be a requirement that if billing is activated for a community, the new owner must also have their Stripe settings activated?

**Comment by momothereal** [comment]
PR: #3863 

Should the transfer flow be:

Selecting the member from the list, clicking the 'cog', and selecting a (new) `Owner` option
![image](https://user-images.githubusercontent.com/6775216/44815774-73c8d180-abae-11e8-895a-5cc5796761e6.png)


**OR**

Clicking on a `Transfer Ownership` button in the settings, selecting the member, confirmation

![image](https://user-images.githubusercontent.com/6775216/44815860-ad014180-abae-11e8-8e7c-c940968cb989.png)

**Comment by brianlovin** [comment]
I think it should be the second flow. Transfer ownership -> choose user -> confirm

**Comment by brianlovin** [comment]
> It could be a requirement that if billing is activated for a community, the new owner must also have their Stripe settings activated?

The billing settings are per community, not per user. Not sure exactly how this will play out but I'll have a think on it.

> Since there would be a confirmation prompt for the new owner, this could be a requirement when accepting.

For acceptance flows we'll definitely need notifications + emails
- notification of request to transfer ownership
- accepted request
- rejected request

**Comment by momothereal** [comment]
@brianlovin Perfect, I don't have to start over 😄 

Is there some place in the code base where you can "select members"? Some sort of list or auto-completing text field would work.

As for billing, my idea is that when accepting the transfer, the new owner would be prompted to setup new billing settings for the community (they wouldn't be allowed to transfer without it). I can see why this is tricky so I'll leave the flow up to you.

**Comment by brianlovin** [comment]
> I can see why this is tricky so I'll leave the flow up to you.

Just to set expectations, I might not be able to get to this for a bit - but agree, I will probably be the best person suited to wrapping this up on the billing side :)
